packages/camera/GUIDE.md

```diff
@@ -0,0 +1,87 @@
+## Guide to understanding the Flutter Camera plugin for Developers
+
+The camera app aims to be a stable plugin for flutter projects.
+The code currently is enabling video capture on both platforms iOS and Android.
+
+
+This guide is in 3 parts:
+
+* Example App
+* Plugin  
+* Native Code
+
+
+### Example App
+
+A demo app to showcase the plugin with all its features located in `example/lib/main.dart`
+
+#### Features
+
+* ResolutionPreset: capture quality of the preview (low medium high) (not available in UI)
+* CameraLensDirection: choose from a list of available cameras ( front back external )
+* capture: save a JPEG image from the chosen camera view with the max available
+  pixel for that camera
+* videostart: Start video capture from the chosen camera view with the chosen `ResolutionPreset`
+* videostop: Stop the video capture and output the file path of the saved video
+
+
+### Plugin
+
+The plugin essentially has one file in the `lib/camera.dart` which which provides
+all the _features_
+
+
+#### Channels
+
+There are two channels communicating with the native code,
+
+* MethodChannel: Transfer of state information, data strings, controls, triggers, calls etc..
+* EventChannel: Transfer of image data stream, tagged with `textureId`
+
+#### Method Calls
+
+* `init` : initialise the respective native cams and reset set them to their original states
+* `list` : list all available cams
+* `create` : create a camera object with the selected camera and resolution preset
+          also start a stream with the event channel with `textureId`. starts the
+          image data stream display too
+* `start` : Start the image data stream display in the UI
+* `stop` :  Stops the image data stream display in the UI
+* `capture` : Capture the JPEG image by sending the image path to be saved
+* `videostart` : Trigger start of the video Recorder variable to capture
+              video to a file ( requires work)
+* `videostop` : Stop the camera from recording, send the file path of the saved file
+              back and dispose the camera ( requires work)
+* `dispose` : Dispose and reset the camera that was initialised
```

Comment:
@nisdis we need to update this names this type of documentation is a double edge sword

Reply:
@luigi-agosti yes i do agree, I am planning to remove the this doc. I had planned to have this file only till the plugin got more stable and worked on both iOS and Android.
We can remove this file.